File: src/main/java/com/example/probation/event/RegistrationListener.java
package com.example.probation.event;

import com.example.probation.service.TokenService;
import lombok.RequiredArgsConstructor;
import org.springframework.context.ApplicationListener;
import org.springframework.stereotype.Component;

@RequiredArgsConstructor
@Component
public class RegistrationListener implements ApplicationListener<OnRegistrationCompleteEvent> {
    private final TokenService tokenService;

    @Override
    public void onApplicationEvent(OnRegistrationCompleteEvent event) {
        this.confirmRegistration(event);
    }

    public void confirmRegistration(OnRegistrationCompleteEvent event) {
        tokenService.confirmRegistration(event);
    }
}

File: src/main/kotlin/com/example/probation/event/RegistrationListener.kt
package com.example.probation.event

import com.example.probation.service.TokenService
import org.springframework.context.ApplicationListener
import org.springframework.stereotype.Component

@Component
class RegistrationListener(
    private val tokenService: TokenService
) : ApplicationListener<OnRegistrationCompleteEvent> {
    override fun onApplicationEvent(event: OnRegistrationCompleteEvent) {
        event.user?.let {
            tokenService.confirmRegistration(it)
        }
    }
}
